src/nvim/api/buffer.c

```diff
@@ -467,7 +467,13 @@ void buffer_insert(Buffer buffer,
                    ArrayOf(String) lines,
                    Error *err)
 {
-  buffer_set_line_slice(buffer, lnum, lnum, true, false, lines, err);
+  linenr_T nlines = buffer_line_count(buffer, err);
```

Comment:
Use `Integer` here, not `linenr_T`. Otherwise you will need one (here) or two (here and in condition) casts to pass travis tests. Also note that `buffer_set_line_slice` accepts Integer as well, though I think it will not complain because `sizeof(Integer) >= sizeof(linenr_T)` always.

And I think that you need to abort if `buffer_line_count` returned zero (== aborted with error, but checking for `err->set` is rather uncommon for some reason).